fog/view/server/src/shard_responses_processor.rs

```diff
@@ -48,13 +49,13 @@ pub fn process_shard_responses(
         // We did not receive a query_response for this shard.Therefore, we need to:
         //  (a) retry the query
         //  (b) authenticate with the Fog View Store that returned the decryption_error
+        let store_uri = FogViewStoreUri::from_str(response.get_fog_view_store_uri())?;
         if response.has_decryption_error() {
             shard_clients_for_retry.push(shard_client);
-            let store_uri =
-                FogViewStoreUri::from_str(&response.get_decryption_error().fog_view_store_uri)?;
             view_store_uris_for_authentication.push(store_uri);
         } else {
-            new_query_responses.push(response.take_query_response());
+            let store_responder_id = ResponderId::from_str(&store_uri.to_string())?;
```

Comment:
Responder IDs are kind of overloaded, but they tend to follow the format: `host:port-<hash>`, so putting a URL into a `ResponderId` structure is going to make it harder to add better validation to them.

I'd just use a `String` here, TBH.